public/angular-app/addins/addins-controller.js

```diff
@@ -48,141 +43,35 @@ function AddinsController(AddinsFactory, UtilityService) {
-                                switch (detailItem.value){
-                                    case 'Never':
-                                        pluginDetail.never = 1;
-                                        break;
-                                    case 'Always':
-                                        pluginDetail.always = 1;
-                                        break;
-                                    case 'ThisSessionOnly':
-                                        pluginDetail.thisSessionOnly = 1;
-                                        break;
-                                }
-                                addinManagerDetails[detailItem.name] = pluginDetail;
-                            }
-                        }
-                    }
-                }
-                return sums;
-            } else {
-                if(entry.pluginName === item.name
-                    && entry.revitVersion === vm.SelectedYear
-                    && isOfficeMatch(entry.office, vm.SelectedOffice)){
-                    sums[entry.user] = (sums[entry.user] || 0) + 1;
-                }
-                return sums;
-            }
-        }, {});
-
-        vm.AddinManagerStats = Object.keys(addinManagerDetails).map(function (item) {
-            return addinManagerDetails[item];
-        });
-
-        var totals = getTotals(output);
-        totals.sort(function(a,b){
-            return a.count - b.count;
-        }).reverse(); // sorted by count
-
-        vm.UserDetails = totals;
-        vm.SelectedPlugin = item.name;
-    };
-
-    //endregion
-
-    //region Utilities
-
-    /**
-     * Utility used to convert data totals to proper format
-     * @param data
-     * @returns {Array}
-     */
-    function getTotals(data){
-        var list = [];
-        for (var k in data){
-            if(data.hasOwnProperty(k)){
-                list.push({name: k, count: data[k]});
-            }
+        var office;
+        if (vm.SelectedOffice !== 'All') {
+            var matchedOffice = vm.officeFilters.find(function (filter) { return filter.name == vm.SelectedOffice; });
+            office = matchedOffice.code;
         }
-        return list;
-    }
+        AddinsFactory.
+            getUsersOfPlugin(item.name, vm.SelectedYear, office).then(function(response){
+                var totals = response.data;
+                totals.sort(function(a,b){
+                    return a.count - b.count;
+                }).reverse(); // sorted by count
+                vm.UserDetails = totals;
+                vm.SelectedPlugin = item.name;
+            });
 
-    /**
-     *
-     * @param office
-     * @param filter
-     * @returns {boolean}
-     */
-    function isOfficeMatch(office, filter){
-        if(filter === 'All') return true;
-        if(filter.constructor === Array){
-            return filter.indexOf(office) > -1;
-        }else {
-            return office === filter;
+        if (item.name === 'AddinManager') {
```

Comment:
I think this needs an else statement that will set the `vm.AddinManagerStats = []` so that when user clicks on something else than addin manager the addin manager details charts gets removed.